shapely/geometry/geo.py

```diff
@@ -38,6 +39,11 @@ def shape(context):
         return MultiLineString(ob["coordinates"])
     elif geom_type == "multipolygon":
         return MultiPolygon(ob["coordinates"], context_type='geojson')
+    elif geom_type == "geometrycollection":
+        geoms = []
+        for g in ob.get("geometries", []):
+            geoms.append(shape(g))
```

Comment:
Can you please use a list comprehension instead of appending to an empty list? I expect that will be faster.